lib/galaxy/dependencies/pipfiles/default/Pipfile

```diff
@@ -83,6 +83,7 @@ cloudbridge = "*"
 social_auth_core = {version = "==3.1.0+gx0", extras = ['openidconnect']}
 cloudauthz = "==0.6.0"
 gxformat2 = "*"
+refgenconf = "==0.7.0"
```

Comment:
Is the strict pin necessary?

Reply:
Depends on the stability of the refgenconf python API. The last update from `0.6.2` to `0.7.0` was not completely backwards compatible, and would have broken somethings if the version auto-bumped.

Safest to pin, but perhaps @nsheff can weigh in?

Reply:
I'd suggest making it `refgenconf = "==0.7.0"` or `refgenconf = ">=0.7.0"`, if possible. You're right, you can't go any lower with the version number to use the functionality needed here, but future versions should be backwards compatible with respect to that.

Reply:
Is the only question regarding backwards compatibility related to the asset paths returned for hisat2_index and bwa_index? 

If so, if we update these, then it should be good. We'll try to keep everything backwards-compatible, but we consider these "bug fixes" on the recipes since the seek keys should return what is actually used by the tool. 

will it be possible to change the pin the future? So,  the pin might be prudent at this point, but could be unpinned after we're sure things aren't changing anymore. I *think* we're there after this change, but I may be wrong...

Reply:
also -- probably want to pin to `0.9.1`, since that will include the new seek keys you requested.

Reply:
@nsheff yes, we can change the pin later. But concern here would be keeping the definitions of the data tables, which a user could have deployed as-is here, with an update to a seek_key would cause a mismatch for that user during upgrade.

`0.9.1` has the changes to hisat2_index and bwa_index?

Reply:
To clarify our pinning process, normally we leave most Python dependencies unpinned in this `Pipfile`, and every couple of months generate from it a `requirements.txt` file with the actual pinnings (including the dependencies' dependencies) using `pipenv`.

Reply:
wait -- I was confused, sorry -- `0.9.1` is for `refgenie`, not `refgenconf`. you are not relying on refgenie here, only on refgenconf. So, ignore that comment. `0.7.0` is correct, that's the latest version.

Refgenconf does not change with the update to the seek keys -- those changes happen in refgenie. So, what matters will actually be what version of refgenie you used to create the assets. Right now, you haven't got any way to *build* assets in Galaxy yet, right? So, users are only going to be using `pull` for assets. In that case, fixing these seek keys is not actually a change to any software, it's just a matter of fixing the actual assets on the server.

I've just made a change to the building component ([here](https://github.com/databio/refgenie/pull/172), and we'll be able to deploy these changes to the server shortly.

Anyway -- what I'm trying to say is the mismatches will not be caused by refgenconf changes -- it's all going to be remote anyway. So the pin doesn't help you with that.

Reply:
For Galaxy driven creation, the building will likely come from a tool(s) added to the toolshed that uses refgenie, which can be added outside of the core Galaxy release cycle. 

I imagine users may want to create their own assets on the command line and load them into their Galaxy instance. 

But it sounds like we can consider the previous asset pull or creation in refgenie to be slightly bugged, with fixes applied or pending. So we should be safe to change our two data table definitions to match what is expected by the underlying executable. Cool, I'll try to find some time to remove, pull, and retest.